### PR TITLE
New Lavaland ruin with 3 ghost roles - Lost Veterinarian 

### PR DESCRIPTION
**Pull request by Zergspower**
## About The Pull Request

The Lost Veterinarian ruin is an RP-focused one, gambling on finding station crew to help them learn anything about how their clinic suddenly landed there after they were in stasis. Choose your own adventure here but stick with your coworkers! If you decide to venture out you'll have very little to defend yourself.

Changes of course can occur during the first phase of this, and i'm certain something in the code is jank and needs tweaking. But the role is definitely a fun one, and depending on how things go i'll adjust and balance along the way. Currently speaking there's a handful of medkits and that's about it, unless you want to build an aquarium by looting that is.

## How This Contributes To The Skyrat Roleplay Experience

One of my favorite characters to ever come into being was born from a similar ruin on another server, in a sense it's an 'easier' version of the Hermit but they will have no materials outside of what a clinic would have. Relying instead on interactions from Miners to tell them what the hell is going on, what is this place, what year is this, who are you etc.

The general rules at the writing of this post are the following:
- They've no knowledge of where they are and have no corporate affiliation with either Interdyne or NT
- They've no ID/headsets so contacting them after locating them would be done face to face
- The facility can be entered and exited regardless if any Vets are there, however don't grief the facility by stealing everything and/or destroying it
- Don't take the surgery duffel kit , toolbox or metal sheets

The Lost Veterinarian
--
Lavaland_vet

## Changelog

:cl:
add: Adds a new ghost_spawn ruin, The lost Veterinarian! 
/:cl:

### CODE REVIEW

**Comment by Jolly-66** [comment]
Since this is for Lavaland, I would STRONGLY recommend putting in generic ash turf so, when map gen loaded, there isn't a horrible chance of lava just being all over this.

**Review by Jolly-66 — CHANGES_REQUESTED** [review]
I'm on mobile and don't know how to line comment but you forgot to include the file you added to the DME.

File: code/__DEFINES/role_preferences.dm [hunk 1]
```diff
@@ -107,7 +107,9 @@
 #define ROLE_BATTLECRUISER_CREW "Battlecruiser Crew"
 #define ROLE_BATTLECRUISER_CAPTAIN "Battlecruiser Captain"
 #define ROLE_VENUSHUMANTRAP "Venus Human Trap"
-
+// SKYRAT EDIT START
+#define ROLE_LOST_VETERINARIAN "Lost Veterinarian"
```
Comment (Zonespace27) [inline code/__DEFINES/role_preferences.dm]: modularize this

File: code/modules/jobs/job_types/spawner/lost_veterinarian.dm [hunk 2]
```diff
@@ -0,0 +1,3 @@
+/datum/job/lost_veterinarian
```
Comment (Zonespace27) [inline code/modules/jobs/job_types/spawner/lost_veterinarian.dm]: modularize this + tick it in the dme

File: code/modules/mob_spawn/ghost_roles/mining_roles.dm [hunk 3]
```diff
@@ -313,6 +313,9 @@
 	implants = list(/obj/item/implant/weapons_auth)
 	id_trim = /datum/id_trim/syndicom/skyrat/interdyne //SKYRAT EDIT
 
+// SKYRAT EDIT
```
Comment (Zonespace27) [inline code/modules/mob_spawn/ghost_roles/mining_roles.dm]: ?

File: modular_skyrat/modules/mapping/code/lavaland_ruin_code.dm [hunk 4]
```diff
@@ -81,6 +92,20 @@
 	uniform = /obj/item/clothing/under/syndicate/skyrat/tactical
 	suit = /obj/item/clothing/suit/hooded/wintercoat/syndicate
 
+/datum/outfit/job/Lavaland_Vet
```
Comment (Zonespace27) [inline modular_skyrat/modules/mapping/code/lavaland_ruin_code.dm]: ew, don't capitalize typepaths

Comment (Zonespace27) [inline modular_skyrat/modules/mapping/code/lavaland_ruin_code.dm]: also as i said in discord does this really need to be a `job/` subtype

File: modular_skyrat/modules/mapping/code/lavaland_ruin_code.dm [hunk 5]
```diff
@@ -17,6 +17,17 @@
 	you_are_text = "You are a syndicate science technician, employed in a top secret research facility developing biological weapons."
 	outfit = /datum/outfit/lavaland_syndicate/ice
 
+/obj/effect/mob_spawn/ghost_role/human/lost_veterinarian
+	name = "Lost Veterinarian"
+	you_are_text = "You're a doctor waking up in a land unfamiliar to you, searing heat surrounds your former veternarian clinic."
+	flavour_text = "Unfortunately you have no idea how you got here, if your customers are still alive or even what year it is."
```
Comment (Zonespace27) [inline modular_skyrat/modules/mapping/code/lavaland_ruin_code.dm]:
```suggestion
	flavour_text = "Unfortunately you have no idea how you got here, if your customers are still alive, or even what year it is."
```
oxford comma

File: modular_skyrat/modules/mapping/code/lavaland_ruin_code.dm [hunk 6]
```diff
@@ -17,6 +17,17 @@
...
+	important_text = "This is a make your own journey ghost-role, it is allowed to go to the station. You aren't mean to attack anyone directly, you're just a vet after all."
```
Comment (Zonespace27) [inline modular_skyrat/modules/mapping/code/lavaland_ruin_code.dm]:
```suggestion
	important_text = "This is a make your own journey ghost-role, it is allowed to go to the station. You aren't meant to attack anyone directly, you're just a vet after all."
```
the second sentence feels entirely unneeded

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/mapping/code/lavaland_ruin_code.dm (hunk 4 above)
Comment (Zergspower) [reply modular_skyrat/modules/mapping/code/lavaland_ruin_code.dm]: weird... that's .. not what it is currently

**Comment by Zergspower** [comment]
> Since this is for Lavaland, I would STRONGLY recommend putting in generic ash turf so, when map gen loaded, there isn't a horrible chance of lava just being all over this.

I did not think of that

File: modular_skyrat/modules/mapping/code/lavaland_ruin_code.dm (hunk 6 above)
Comment (Zergspower) [reply modular_skyrat/modules/mapping/code/lavaland_ruin_code.dm]: think so? Idea was just to clear the baseline up front - it do come off a lil aggressive

File: code/__DEFINES/role_preferences.dm [hunk 7]
```diff
@@ -108,7 +108,6 @@
 #define ROLE_BATTLECRUISER_CAPTAIN "Battlecruiser Captain"
 #define ROLE_VENUSHUMANTRAP "Venus Human Trap"
 
-
```
Comment (Zonespace27) [inline code/__DEFINES/role_preferences.dm]: don't remove

File: modular_skyrat/modules/mapping/code/lavaland_ruin_code.dm [hunk 8]
```diff
@@ -17,6 +17,15 @@
 	you_are_text = "You are a syndicate science technician, employed in a top secret research facility developing biological weapons."
 	outfit = /datum/outfit/lavaland_syndicate/ice
 
+/obj/effect/mob_spawn/ghost_role/human/lost_veterinarian
+	name = "Lost Veterinarian"
```
Comment (Zonespace27) [inline modular_skyrat/modules/mapping/code/lavaland_ruin_code.dm:21]: add a `prompt_name`

File: modular_skyrat/modules/mapping/code/lavaland_ruin_code.dm [hunk 9]
```diff
@@ -81,6 +90,14 @@
 	uniform = /obj/item/clothing/under/syndicate/skyrat/tactical
 	suit = /obj/item/clothing/suit/hooded/wintercoat/syndicate
 
+/datum/outfit/lavaland_vet
```
Comment (Zonespace27) [inline modular_skyrat/modules/mapping/code/lavaland_ruin_code.dm:94]: Blueshift: /datum/unit_test/outfit_sanity
Lost Veterinarian's backpack_contents are invalid! Couldn't add /obj/item/storage/box/survival/medical to backpack.

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/__DEFINES/role_preferences.dm (hunk 7 above)
Comment (Zergspower) [reply code/__DEFINES/role_preferences.dm]: I'm confused i was told to remove that?

Comment (Zonespace27) [reply code/__DEFINES/role_preferences.dm]: don't remove the random line of whitespace

Comment (Zergspower) [reply code/__DEFINES/role_preferences.dm]: OH I didnt realize i did that

**Review by Paxilmaniac — CHANGES_REQUESTED** [review]
I question the need for having more ghost roles than we already do, but past that I have nitpicks about the map.

Overalls, it's very boxy, and the design is pretty... bland? It doesn't seem like there's much to do and I can forsee the place just being abandoned as the doctors take their spawn and just leave for someplace actually interesting after gathering all the supplies.

image
Why is there an anesthetic tank just sitting in the open in the lobby?

image
Again the anesthetic tank.

image
A styling suggestion, replace the the center window of the five with a wall.

image
image
I don't see the need for a fishing portal and rod setup, remove it and move the air tank connected to the lobby into here so you can do away with that little closet.

image
Make these normal sutures.

image
This whole back area should be access restricted in some way, including adding doors that the big hallway on the left so people can't just stroll in from the lobby without someone with access letting them in... or them just breaking in.

image
I think just a plant pot in the bathroom is more fitting instead of this.

image
The left side of this room is very empty, perhaps you could find something to put here to fit the space?

**Comment by YakumoChen** [comment]
kind of an eh port / revamp of an already very eh role. Of the servers I've seen (mostly citadel) that had vet, almost nobody ever played or, or it was only used as a way to get other roles like lavaland xenos going. The concept just really isn't interesting except maybe a couple times.

The ruin itself was just a practical joke from a coder, and the building makes no real sense to belong on lavaland in the first place.

We have far  more immersive roles with the same idea with the likes of Port Tarkon or Charlie Station, or hermit and ash lizards if you want to have the real lavaland innawoods experience.

**Comment by YakumoChen** [comment]
you also didn't add the sleeper spawns for the (sentient) animals which was kind of part of the entire point.

**Comment by Zergspower** [comment]
> you also didn't add the sleeper spawns for the (sentient) animals which was kind of part of the entire point.

Sentient animals what? This wasn't a remake of the TG version - Ashies, Tarkon and Charlie don't get the freedom to interact with the station that this provides

**Comment by YakumoChen** [comment]
> > you also didn't add the sleeper spawns for the (sentient) animals which was kind of part of the entire point.
> 
> Sentient animals what? This wasn't a remake of the TG version - Ashies, Tarkon and Charlie don't get the freedom to interact with the station that this provides

Aside from the language barrier ashies seem to have plenty of freedom to contact the station or miners. 

The theme of the ruin here just doesn't make sense. It fails the question of "why would this be here?". All the fauna of lavaland are explicitly hostile. 

Also concerned about the new policy burden of having an all access ghost role ruin that is seemingly innocuous but you're not allowed to take mundane items?

If you wanted a new ghost ruin on lavaland (among the 4-5 already in place), there are plenty of more immersive avenues to explore. Disgraced/ruined cults, beefed up hermit encampments, russians?

**Comment by Zergspower** [comment]
> > > you also didn't add the sleeper spawns for the (sentient) animals which was kind of part of the entire point.
> > 
> > 
> > Sentient animals what? This wasn't a remake of the TG version - Ashies, Tarkon and Charlie don't get the freedom to interact with the station that this provides
> 
> Aside from the language barrier ashies seem to have plenty of freedom to contact the station or miners.
> 
> The theme of the ruin here just doesn't make sense. It fails the question of "why would this be here?". All the fauna of lavaland are explicitly hostile.
> 
> Also concerned about the new policy burden of having an all access ghost role ruin that is seemingly innocuous but you're not allowed to take mundane items?
> 
> If you wanted a new ghost ruin on lavaland (among the 4-5 already in place), there are plenty of more immersive avenues to explore. Disgraced/ruined cults, beefed up hermit encampments, russians?

Ashies are heavily limited strictly from a language/lore perspective, I did give thought to russians/nri but couldnt really find another reason other than the translocation aspect this ruin has. Adding to that - we'd then have Syndies 2.0 with the NRI/Russians so idk.   I'm wanting to try the angle of an open aspect role, go join the syndies, get rescued by miners and go to the station and build a scenario around your past. Tbh the beach and pod spawns probably just need to get removed in my next ruin pass as they're never populated given their restrictive rules

**Comment by YakumoChen** [comment]
pods are fine, they're a tutorial / chill botany role tbh
beach bums I thought already had the spawns removed? It's about as pointless to play as this ruin tbh

**Comment by tf-4** [comment]
Seeing a lack of interest from other maintainers and I'm not very interested myself. It just doesn't have enough meat to be a ghost role I'm keen on adding. If you can think of a way to give it more of a gameplay loop I'm happy to reopen.

**Comment by Zergspower** [comment]
> Seeing a lack of interest from other maintainers and I'm not very interested myself. It just doesn't have enough meat to be a ghost role I'm keen on adding. If you can think of a way to give it more of a gameplay loop I'm happy to reopen.

Would've preferred this be drafted, i'm about to do a sweep of removals

**Pull request closed without merging**